src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/controllers/api/DocumentController.kt

```diff
@@ -14,114 +15,56 @@ import java.io.File
 class DocumentController(context: ActionContext,
                          private val fileSystem: FileSystem,
                          config: Config,
-                         private val repo: DocumentRepository): Controller(context)
+                         private val repo: DocumentRepository) : Controller(context)
 {
     constructor(context: ActionContext) :
             this(context,
                     Files(),
                     AppConfig(),
                     OrderlyDocumentRepository())
 
-    private val topLevelFolder = fileSystem.getAbsolutePath(config["documents.root"])
+    private val documentsRoot = fileSystem.getAbsolutePath(config["documents.root"])
 
-    fun refreshDocuments(): String {
+    fun refreshDocuments(): String
+    {
         val allDocs = repo.getAllFlat()
-
+        val root = DocumentDetails("root", documentsRoot, null)
         val unrefreshedDocs = allDocs.toMutableList()
-
-        val topLevelDir = unrefreshedDocs.find{it.path == relativePath(topLevelFolder)}
-        if ( topLevelDir == null)
-        {
-            repo.add(relativePath(topLevelFolder), documentName(topLevelFolder), false, null)
-        }
-        else
-        {
-            unrefreshedDocs.remove(topLevelDir)
-        }
-        refreshDocumentsInDir(topLevelFolder,  unrefreshedDocs)
+        refreshDocumentsInDir(root, unrefreshedDocs)
 
         //Hide all known docs which were not found
-        unrefreshedDocs.filter{ it.show }.forEach{ repo.setVisibility(it, false) }
+        unrefreshedDocs.forEach { repo.setVisibility(it, false) }
 
         return okayResponse()
     }
 
-    private fun refreshDocumentsInDir(absolutePath: String,
+    private fun refreshDocumentsInDir(document: DocumentDetails,
                                       unrefreshedDocs: MutableList<Document>)
     {
-        val files = fileSystem.getChildFiles(absolutePath)
-        val folders = fileSystem.getChildFolders(absolutePath)
-
-        refreshDirChildren(absolutePath, files, false, unrefreshedDocs)
-        refreshDirChildren(absolutePath, folders, true, unrefreshedDocs)
+        val files = fileSystem.getAllChildren(document.absolutePath, documentsRoot)
+        refreshDirChildren(document, files, unrefreshedDocs)
     }
 
-    private fun refreshDirChildren(dirAbsolutePath: String,
-                                   children: List<String>,
-                                   areFolders: Boolean,
+    private fun refreshDirChildren(parent: DocumentDetails,
```

Comment:
The only reason I pulled this out as a separate function was to put the file and folders results from Files through it one by one. Since that's gone now, maybe this doesn't to be a separate function any more.